doc/api/process.markdown

```diff
@@ -286,6 +286,10 @@ event and that writes can block when output is redirected to a file (although
 disks are fast and operating systems normally employ write-back caching so it
 should be a very rare occurrence indeed.)
 
+Note that on `process.stdout` and `process.stderr`, the callback passed to 
+`stream.write` might be called before all written data is flushed completely. 
+This can result in data loss if Node.js is ended prematurely using `process.exit`.
```

Comment:
This is incorrect.  What you should convey here is that the callback can fire before the receiving end has received all data because it's still in transit in a kernel buffer.

The other caveat is that calling `process.exit` before callbacks have fired means the data may not have been handed off to the operating system yet.  The imperative word here is 'may' - it may or may not have been handed off.  It's not until the callback fires that you know for sure.

In case it isn't clear, the flow is `node -> kernel -> other`, where `other` is either a tty, a file or another process reading from a pipe.  Calling `process.exit` when pending data is still in node land means the data is irretrievably lost.  When it's been handed off to the kernel, it's unspecified what happens - it depends on the operating system and the phase of the moon.